### PR TITLE
ci: fix pipeline

### PR DESCRIPTION
**Pull request by weichweich**
## fixes KILTProtocol/ticket#1695

* You can [check that the yaml is valid in gitlab](https://gitlab.com/kiltprotocol/mashnet-node/-/ci/editor?branch_name=aw-image-pipeline&tab=2)

## Checklist:

- [ ] I have verified that the code works
  - [ ] No panics! (checked arithmetic ops, no indexing `array[3]` use `get(3)`, ...)
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: .maintain/push-image.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+source_tag=$1
+target_tag=$2
+
+# publish to docker hub
+docker tag kiltprotocol/mashnet-node:$source_tag ${DOCKER_HUB_STANDALONE}:$target_tag
+docker tag kiltprotocol/kilt-node:$source_tag ${DOCKER_HUB_PARACHAIN}:$target_tag
```
Comment (Diiaablo95) [inline .maintain/push-image.sh:8]: Isn't `kiltprotocol/kilt-node` = `${DOCKER_HUB_PARACHAIN}`?

File: .maintain/build-image.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+target_tag=$1
+
+# build parachain image and standalone image
+docker build --cache-from $AWS_REGISTRY/kilt-parachain/collator:$target_tag --build-arg NODE_TYPE=kilt-parachain -t kiltprotocol/kilt-node:$target_tag .
```
Comment (Diiaablo95) [inline .maintain/build-image.sh:6]: I think we should replace all occurrences of `kiltprotocol/*` with the relative env variable.

File: .maintain/push-image.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+docker push ${DOCKER_HUB_STANDALONE}:$target_tag
+docker push ${DOCKER_HUB_PARACHAIN}:$target_tag
+
+# publish to AWS
+docker tag kiltprotocol/mashnet-node:$source_tag $AWS_REGISTRY/kilt/prototype-chain:$target_tag
```
Comment (Diiaablo95) [inline .maintain/push-image.sh:14]: Same here for the image name, that should be replaced with the respective env variable.

**Review by Diiaablo95 — COMMENTED** [review]
Looks definitely cleaner! Perhaps I would make more use of the env variables. Plus, a question: I don't see any occurrence of the `latest` tag anywhere. Is that somehow to be triggered manually?

File: .maintain/build-image.sh (hunk 2 above)
Comment (weichweich) [reply .maintain/build-image.sh:6]: I'll rename that. The idea is, that the build script builds the images with a name X which doesn't really matter how it's called.
The publish script tags X to the docker and AWS names, and pushes those two.

**Comment by weichweich** [comment]
> I don't see any occurrence of the `latest` tag anywhere. Is that somehow to be triggered manually?

The magic happens [in the gitlab ci file](https://github.com/KILTprotocol/mashnet-node/pull/328/files#diff-037ea159eb0a7cb0ac23b851e66bee30fb838ee8d0d99fa331a1ba65283d37f7R56-R58).

* if branch -> create a tag with short commit SHA
* if branch -> push images with tag `latest-{branch-name}`
* if tag -> push images with tag `{tag-name}`

File: .maintain/push-image.sh (hunk 1 above)
Comment (weichweich) [reply .maintain/push-image.sh]: [1b65854](https://github.com/KILTprotocol/mashnet-node/pull/328/commits/1b65854c84e632b2fedd2b871b97fec90baa135a)

**Comment by Diiaablo95** [comment]
Yes, I see that, and as far as I can see, we won't have a `latest` Docker tag unless we push a `latest` git tag, right? I think GitHub releases can only be associated with one git tag, which would be the runtime version, so I am not sure how we can associated the latest release with the `latest` Docker tag.

**Comment by weichweich** [comment]
> Yes, I see that, and as far as I can see, we won't have a `latest` Docker tag unless we push a `latest` git tag, right? I think GitHub releases can only be associated with one git tag, which would be the runtime version, so I am not sure how we can associated the latest release with the `latest` Docker tag.

Ah now i got you! the only latest thing! :joy: yes i forgot that!

**Review by Diiaablo95 — APPROVED** [review]
Grrrrrreat! 😎

**Comment by weichweich** [comment]
let's see if it works. 😓

**Pull request merged**